pandas/tools/util.py

```diff
@@ -61,6 +61,13 @@ def to_numeric(arg, errors='raise'):
         - If 'raise', then invalid parsing will raise an exception
         - If 'coerce', then invalid parsing will be set as NaN
         - If 'ignore', then invalid parsing will return the input
+    downcast : str, numpy.dtype, or Python dtype, default None
```

Comment:
not sure I see a problem - if I specify u it then you simply try to progressively downcast until it works starting with uint8,uint16,uint32,uint64 (prob easiest to check do for negatives first )

so you will have to dispatch between the compacting routine and the downcast depending on the input

eg if u get int,signed,unsigned,float then u r compacting otherwise you astype to the specified dtype 

and reject anything else (and any non int/float dtypes)